### ISSUE TITLE
Jack support on Mac and Windows

### ISSUE THREAD

**Issue by hlolli** [csound/csound#1147]
The current behaviour on Mac is the same as for windows, jack isn't found and it will default to a dummy module and no jack client is created on the jack server. I'll leave the logs for macintosh, but I'd be happy to see cross platform support for jack audio, behaving essentially the same as on linux (just refering to the audio module output, not the jack opcodes).

```
jackd -d coreaudio
jackdmp 1.9.13

Copyright 2001-2005 Paul Davis and others.

Copyright 2004-2016 Grame.

Copyright 2016-2019 Filipe Coelho.

jackdmp comes with ABSOLUTELY NO WARRANTY

This is free software, and you are welcome to redistribute it

under certain conditions; see the file COPYING for details

JACK server starting in realtime mode with priority 10

self-connect-mode is "Don't restrict self connect requests"

Default input and output devices are not the same !!

Cannot open default device in duplex mode, so aggregate default input and default output

CreateAggregateDeviceAux : devices do not share the same clock!! clock drift compensation would be needed...

Separated input = 'Soundflower (64ch)' 

Separated output = 'Soundflower (2ch)' 

CoreAudio driver is running...
```
```

csound -odac HelloWorld.csd -+rtaudio=jack

time resolution is 1000.000 ns

0dBFS level = 32768.0

--Csound version 6.13 beta (double samples) May 27 2019

[commit: 0591fffdb46b222cf78ea068ee3665d9f5cc310c]

libsndfile-1.0.28

UnifiedCSD:  HelloWorld.csd

STARTING FILE

Creating options

Creating orchestra

closing tag

Creating score

Elapsed time at end of orchestra compile: real: 0.010s, CPU: 0.004s

sorting score ...

	... done

Elapsed time at end of score sort: real: 0.011s, CPU: 0.004s

graphics suppressed, ascii substituted

0dBFS level = 32768.0

orch now loaded

audio buffered in 1024 sample-frame blocks

 unknown rtaudio module: 'jack', using dummy module

rtaudio: dummy module enabled

writing 1024 sample blks of 64-bit floats to dac

SECTION 1:

new alloc for instr 1:

B  0.000 ..  1.000 T  1.000 TT  1.000 M:   8192.0

Score finished in csoundPerform().

inactive allocs returned to freespace

end of score.		   overall amps:   8192.0

	   overall samples out of range:        0

0 errors in performance

Elapsed time at end of performance: real: 1.014s, CPU: 0.012s

44 1024 sample blks of 64-bit floats written to dac
```

**Comment by vlazzarini** [comment]
It works alright here on the Mac. I can't say for Wndows.

With jack running, Csound connects and plays.

**Comment by stekyne** [comment]
The Windows build doesn't have Jack in it as far as I can remember. Maybe create a work item to include it?

**Comment by vlazzarini** [comment]
Looking at this again, the likelihood is that you don't have librtjack.dylib in your system. Since this is your own build (6.13 has not been released yet), it's up to you to build it. I am closing this, since it's misleading. If you are having difficulty building the plugin, please ask in the Developer's list.
For Windows, a ticket needs to be opened as a feature request/enhancement for that platform.

**Comment by hlolli** [comment]
By providing -DUSE_JACK=1 and  -DJACK_HEADER=/path/to/jack/commons I was able to generate librtjack.dylib. Given that on linux, the cmake is able to find the jack libraries, it seems to be unable to find it on machintosh. I was hesitant to provide JACK_HEADER because I took it for granted to be only needed for jack opcodes (that's the behaviour on linux). But then again, from my jack installation, it did not include jack.h globally on the machintosh I'm running on (it's an emulator), so it's perhaps the only way to compile jack, I will try the same on Windows, it could be that it's supported and I only need to provide cmake flags.
